slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/validator/AttestationValidator.java

```diff
@@ -65,16 +74,21 @@ public boolean sourceGreaterThanTargetEpoch() {
     return false;
   }
 
-  public boolean existsInDatabase() {
+  public boolean alreadyExists() {
     return signedAttestationsDao
         .findMatchingAttestation(handle, validatorId, targetEpoch, signingRoot)
         .isPresent();
   }
 
-  public void insertToDatabase() {
+  public void persist() {
     final SignedAttestation signedAttestation =
         new SignedAttestation(validatorId, sourceEpoch, targetEpoch, signingRoot);
     signedAttestationsDao.insertAttestation(handle, signedAttestation);
+
+    // update the watermark if is otherwise blank
+    if (watermark.isEmpty() || (watermark.get().getSourceEpoch() == null)) {
```

Comment:
Possibly also check that targetEpoch is null. That should be set in pairs but the db allows for them to be set independently.

Reply:
it looks like we can create a check constraint in the db ... but not sure how